File: java/Fix1A.java

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class Fix1A {
    public static void main(String[] args) {
        try {
            String text = new String(Files.readAllBytes(Paths.get(args[0])));
            if (text.getBytes()[text.getBytes().length-1]==26)
                Files.write(Paths.get(args[0]),text.trim().getBytes());
        } catch (final IOException e) {
            e.printStackTrace();
        }
    }
}

File: kotlin/Fix1A.kt
import java.nio.file.Files
import java.nio.file.Paths

public fun main(args: Array<String>){
    val f =Files.readAllBytes(Paths.get(args[0]))
    if(f.last().toInt()==26) 
        Files.write(Paths.get(args[0]),f.copyOfRange(0, f.lastIndex - 2))
}